### PR TITLE
work on C4CA changes

### PR DESCRIPTION
**Pull request by saloni-080601**
work on C4CA pathway

Make a new C4CA pathway

### CODE REVIEW

File: src/components/PathwayExercise/CourseCompletion/CourseCompletionPage.js [hunk 1]
```diff
@@ -6,28 +6,47 @@ import { interpolatePath, PATHS } from "../../../constant";
...
+                      pathwayId: params.pathwayId,
+                    })
+                  )
+                : history.push(
+                    interpolatePath(PATHS.PATHWAY_COURSE_CONTENT, {
+                      courseId: data?.[nextPathwayIndex]?.id,
+                      exerciseId: 0,
+                      pathwayId: params.pathwayId,
+                    })
+                  );
```
Comment (jschanker) [inline src/components/PathwayExercise/CourseCompletion/CourseCompletionPage.js]:
```suggestion
              history.push(
                interpolatePath(PATHS.PATHWAY_COURSE_CONTENT, {
                  courseId: C4CALastPage
                    ? pathwayModule?.[pathwayModuleIndex]?.id
                    : data?.[nextPathwayIndex]?.id,
                  exerciseId: 0,
                  pathwayId: params.pathwayId,
                })
              );
```

File: src/components/PathwayExercise/CourseCompletion/LastCoursePage.js [hunk 2]
```diff
@@ -20,11 +21,13 @@ function LastCoursePage() {
...
-                  pathwayId: params.pathwayId,
-                })
-              );
+              C4CALastPage
+                ? history.push("/c4ca-pathway")
+                : history.push(
+                    interpolatePath(PATHS.PATHWAY_COURSE, {
+                      pathwayId: params.pathwayId,
+                    })
+                  );
```
Comment (jschanker) [inline src/components/PathwayExercise/CourseCompletion/LastCoursePage.js]:
```suggestion
              history.push(
                C4CALastPage
                  ? "/c4ca-pathway"
                  : interpolatePath(PATHS.PATHWAY_COURSE, {
                      pathwayId: params.pathwayId,
                    })
              );
```

**Review by jschanker — COMMENTED** [review]
couple of quick suggestions to start

File: src/components/PathwayCourse/C4CApathway/index.js [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+                      })
+                    }
+                  >
+                    <Card
+                      className={classes.pathwayCard}
+                      elevation={0}
+                      sx={{
+                        ml: 3,
+                        // p: "16px",
+                        mb: isActive ? "0px" : "16px",
```
Comment (jschanker) [inline src/components/PathwayCourse/C4CApathway/index.js]:
```suggestion
                        mb: {xs: 0, sm: "16px"},
```

File: src/components/PathwayCourse/C4CApathway/index.js [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+import { useMediaQuery } from "@mui/material";
+import { breakpoints } from "../../../theme/constant";
+
+function C4CApathway() {
+  const user = useSelector(({ User }) => User);
+  const [pathway, setPathway] = useState([]);
+  const [completedPortion, setCompletedPortion] = useState({});
+  const classes = useStyles();
+  let completedAll = completedPortion?.total === 100;
+  const isActive = useMediaQuery("(max-width:" + breakpoints.values.sm + "px)");
```
Comment (jschanker) [inline src/components/PathwayCourse/C4CApathway/index.js]: Is badlaav ke baad tum yeh line hata saktee ho.
```suggestion
```

File: src/components/PathwayCourse/C4CApathway/index.js [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+
+  // Function to check if a course is unlocked
+
+  // Effect to automatically unlock courses when their percentage reaches 100
+  useEffect(() => {
+    filterCourses &&
+      filterCourses.forEach((item, index) => {
+        const isLastCourse = item.courses.length - 1;
+        item.courses.forEach((course, courseindex) => {
+          if (parseInt(course.completed_portion) === 100) {
```
Comment (jschanker) [inline src/components/PathwayCourse/C4CApathway/index.js]: Yahan `parseInt` kyon zarooree hai lekin neeche nahi (https://github.com/navgurukul/bhanwari-devi/blob/7e8ba4e24b5b498d05e1285b2c6650c32f438ffb/src/components/PathwayCourse/C4CApathway/index.js#L96)? Kyaa back-end string `completed_portion` ke liye bheja raha hai?

File: src/components/PathwayCourse/C4CApathway/index.js [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+  });
+
+  const [unlockedCourses, setUnlockedCourses] = useState([]);
+
+  // Function to check if a course is unlocked
+
+  // Effect to automatically unlock courses when their percentage reaches 100
+  useEffect(() => {
+    filterCourses &&
+      filterCourses.forEach((item, index) => {
```
Comment (jschanker) [inline src/components/PathwayCourse/C4CApathway/index.js:56]:
```suggestion
    filterCourses?.forEach((item, index) => {
```

File: src/components/PathwayCourse/C4CApathway/index.js [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+        const isLastCourse = item.courses.length - 1;
```
Comment (jschanker) [inline src/components/PathwayCourse/C4CApathway/index.js]: Yeh variable ka naam `lastCourseIndex` ko baadalen kyonki yeh Boolean nahi hai.
```suggestion
        const lastCourseIndex = item.courses.length - 1;
```

File: src/components/PathwayCourse/C4CApathway/index.js [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+        item.courses.forEach((course, courseindex) => {
+          if (parseInt(course.completed_portion) === 100) {
+            if (courseindex < isLastCourse) {
```
Comment (jschanker) [inline src/components/PathwayCourse/C4CApathway/index.js]: oopar ke badalte ke saath: 
```suggestion
            if (courseindex < lastCourseIndex) {
```

File: src/components/PathwayCourse/C4CApathway/index.js (hunk 5 above)
Comment (saloni-080601) [reply src/components/PathwayCourse/C4CApathway/index.js]: I remove parseInt

**Pull request merged**